crypto/dilithium/p_pqdsa_asn1.c

```diff
@@ -0,0 +1,257 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/evp.h>
+
+#include <openssl/bytestring.h>
+#include <openssl/err.h>
+#include <openssl/mem.h>
+
+#include "../evp_extra/internal.h"
+#include "../fipsmodule/evp/internal.h"
+#include "../internal.h"
+#include "ml_dsa.h"
+#include "internal.h"
+
+static void pqdsa_free(EVP_PKEY *pkey) {
+  PQDSA_KEY_free(pkey->pkey.pqdsa_key);
+  pkey->pkey.pqdsa_key = NULL;
+}
+
+static int pqdsa_set_priv_raw(EVP_PKEY *pkey, const uint8_t *privkey,
+        size_t privkey_len, const uint8_t *pubkey, size_t pubkey_len) {
+
+  PQDSA_KEY *key = OPENSSL_malloc(sizeof(PQDSA_KEY));
+  if (key == NULL) {
+    return 0;
+  }
+
+  // At time of writing, all |set_priv_raw| and |pqdsa_set_priv_raw|
+  // invocations specify NULL public key. If that changes, we should modify
+  // the conditional below to set the public key on |key|.
+  if (pubkey != NULL) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE);
+    return 0;
+  }
+
+  // NOTE: No checks are done in this function, the caller has to ensure
+  //       that the pointers are valid and |privkey| has the correct size.
+  key->public_key = NULL;
+  key->secret_key = OPENSSL_memdup(privkey, privkey_len);
+  pqdsa_free(pkey);
+  pkey->pkey.pqdsa_key = key;
+  return 1;
+}
+
+static int pqdsa_set_pub_raw(EVP_PKEY *pkey, const uint8_t *in, size_t len) {
+  //generate a fresh pqdsa_key
+  PQDSA_KEY *key = OPENSSL_malloc(sizeof(PQDSA_KEY));
+
+  if (key == NULL) {
+    return 0;
+  }
+
+  // NOTE: No checks are done in this function, the caller has to ensure
```

Comment:
why not?

Reply:
See https://github.com/aws/aws-lc/pull/1963#discussion_r1838610442